packages/calling/calling/src/CallingClient/line/index.ts

```diff
@@ -0,0 +1,461 @@
+  WebexRequestPayload,
+} from '../../common/types';
+import {ICallSettingResponse, ILine, LINE_EVENTS, LineEventTypes, LineStatus} from './types';
+import {
+  LINE_FILE,
+  CALLING_USER_AGENT,
+  CISCO_DEVICE_URL,
+  DISCOVERY_URL,
+  GET_MOBIUS_SERVERS_UTIL,
+  IP_ENDPOINT,
+  KEEPALIVE_UTIL,
+  SPARK_USER_AGENT,
+  URL_ENDPOINT,
+  REGISTRATION_FILE,
+  NETWORK_FLAP_TIMEOUT,
+} from '../constants';
+import log from '../../Logger';
+import {IRegistration} from '../registration/types';
+import {createRegistration} from '../registration';
+import {ClientRegionInfo, ISDKConnector, WebexSDK} from '../../SDKConnector/types';
+import {CallingClientConfig} from '../types';
+import {Eventing} from '../../Events/impl';
+import {LineError} from '../../Errors/catalog/LineError';
+import {METRIC_EVENT, REG_ACTION, METRIC_TYPE} from '../metrics/types';
+import {LOGGER, LogContext} from '../../Logger/types';
+import {handleRegistrationErrors, filterMobiusUris, validateServiceData} from '../../common';
+import SDKConnector from '../../SDKConnector';
+import {ICallManager} from '../calling/types';
+
+export default class Line extends Eventing<LineEventTypes> implements ILine {
+  private webex: WebexSDK;
+
+  private mutex: Mutex;
+
+  private registration: IRegistration;
+
+  private sdkConfig: CallingClientConfig | undefined;
+
+  private sdkConnector: ISDKConnector;
+
+  private callManager: ICallManager;
+
+  public userId: string;
+
+  public clientDeviceUri: string;
+
+  public lineId: string;
+
+  public phoneNumber: string;
+
+  public extension: string;
+
+  public status: LineStatus;
+
+  public sipAddresses: string[];
+
+  public voicemail: string;
+
+  public lastSeen: string;
+
+  public keepaliveInterval: number;
+
+  public callKeepaliveInterval: number;
+
+  public rehomingIntervalMin: number;
+
+  public rehomingIntervalMax: number;
+
+  public voicePortalNumber: number;
+
+  public voicePortalExtension: number;
+
+  private primaryMobiusUris: string[];
+
+  private backupMobiusUris: string[];
+
+  constructor(
+    userId: string,
+    clientDeviceUri: string,
+    lineId: string,
+    phoneNumber: string,
+    extension: string,
+    status: LineStatus,
+    sipAddresses: string[],
+    voicemail: string,
+    lastSeen: string,
+    keepaliveInterval: number,
+    callKeepaliveInterval: number,
+    rehomingIntervalMin: number,
+    rehomingIntervalMax: number,
+    voicePortalNumber: number,
+    voicePortalExtension: number,
+    callManager: ICallManager,
+    mutex: Mutex,
+    config?: CallingClientConfig
+  ) {
+    super();
+    this.userId = userId;
+    this.clientDeviceUri = clientDeviceUri;
+    this.lineId = lineId;
+    this.phoneNumber = phoneNumber;
+    this.extension = extension;
+    this.status = status;
+    this.sipAddresses = sipAddresses;
+    this.voicemail = voicemail;
+    this.lastSeen = lastSeen;
+    this.keepaliveInterval = keepaliveInterval;
+    this.callKeepaliveInterval = callKeepaliveInterval;
```

Comment:
Most of them wil be default value you don't have to pass them in , We can have a config file instead

Reply:
I'm not sure if we need to set them as default value or will it come through config file from DMS. @Kesari3008 can suggest.

Reply:
@meta-dipanshu-sharma as of now we aren't going to contact DMS to get the device configuration. So we don't need to keep dummy data. When the Mobius starts supporting multiple lines (which will take some time), we can add this back

Reply:
lineId, extension, phone number, voicemail are a few properties that will be coming from the config file from DMS but since we don't have that implemented for now, we can leave these values blank. 

Other values like intervals, lastSeen and sipAddresses, can be populated once we get registration response from Mobius.

Reply:
Done